### DISCUSSION TITLE
Question on weak SINDy with ensembing

### DISCUSSION THREAD

**Discussion by scatr** [dynamicslab/pysindy#470] [General]
Hi,

Optimizers like `STLSQ` have the attribute `optimizer.Theta_` which allows you to return the values of the constructed feature library. With weak SINDy, what would be returned in this case if ensembling is also used? Is the returned `Theta_` the library corresponding to the final model fit? Is a history of all fits stored? 

Thanks,
Alasdair

**Comment by Jacob-Stevens-Haas** [comment]
If you an ensemble an STLSQ optimizer, the last value of `Theta_` will be stored (i.e. the samples and features from the final bag).  Also note we have removed `Theta_` from STLSQ, as it was undocumented and didn't seem to serve a huge purpose.  If you want to calculate the weak features using the subdomains from the `fit()`, use test `model.feature_library.transform(data)`

**Reply by scatr** [reply]
Thanks Jacob
